.github/workflows/zxc-build-library.yaml

```diff
@@ -10,94 +10,114 @@ permissions:
-        env:
-          VCPKG_BINARY_SOURCES: clear
-        with:
-          configurePreset: ${{ matrix.preset }}-debug
-          buildPreset: ${{ matrix.preset }}-debug
-
-      - name: CMake Build (Release)
-        uses: lukka/run-cmake@v10
-        env:
-          VCPKG_BINARY_SOURCES: clear
-        with:
-          configurePreset: ${{ matrix.preset }}-release
-          buildPreset: ${{ matrix.preset }}-release
-
-      #      - name: List Directory Structure
-      #        if: ${{ contains(runner.os, 'Windows') && always() }}
-      #        shell: pwsh
-      #        run: tree /F build
-
-      #- name: Start the local node
-      #  if: success()
-      #  run: sudo npx @hashgraph/hedera-local start -d --network local
-
-      #- name: Start CTest suite (Debug)
-      #  run: ctest -C Debug --test-dir build/${{ matrix.preset }}-debug --output-on-failure
-
-      #- name: Start CTest suite (Release)
-      #  run: ctest -C Debug --test-dir build/${{ matrix.preset }}-release --output-on-failure
-
-      - name: Compute Short SHA
-        id: sha
-        run: echo "short=$(echo -n "${{ github.sha }}" | cut -c1-8)" >> $GITHUB_OUTPUT
-
-      - name: Attach Artifact
-        uses: actions/upload-artifact@v3
-        with:
-          name: hapi-library-${{ runner.os }}-${{ steps.sha.outputs.short }}
-          path: package/
-          if-no-files-found: warn
-
-      #- name: Stop the local node
-      #  run: sudo npx @hashgraph/hedera-local stop
\ No newline at end of file
+    build:
+      name: Build
+      # runs-on: ubuntu-latest
+      runs-on:
+        - self-hosted
+        - "${{ matrix.os }}"
+        - xlarge
+        - ephemeral
+      env:
+        HEDERA_NETWORK: localhost
+      strategy:
+        matrix:
+          include:
+            - os: Linux
+              preset: linux-x64
+            # - os: macos-12
+            #   preset: macos-x64
+            # - os: windows-latest
+            #   preset: windows-x64
+      steps:
+          - name: Checkout Code
+            uses: actions/checkout@v3
+            with:
+              submodules: true
+          - name: Install Docker Compose
+            run: |
+              sudo apt-get update
+              sudo apt-get install -y \
+                  ca-certificates \
+                  curl \
+                  gnupg \
+                  lsb-release
+              sudo mkdir -p /etc/apt/keyrings
+              curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+              echo \
+                "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+                $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+              sudo apt-get update
+              sudo apt-get install -y docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
```

Comment:
This step needs to be restricted to Linux only using an `if` clause or needs to be replaced with a universal step such as https://github.com/marketplace/actions/docker-setup-docker which provides WIndows, Linux, and MacOS support. 

I strongly suggest using the universal step from crazymax because we must ultimately support Windows and MacOS as well in these workflows.

Reply:
It seems the version installed via the `crazy-max/ghaction-setup-docker` conflicts with the one that's already globally installed in the runner image. The options that I see are:
* try to develop some workaround to kill and remove the preinstalled version
* update the globally installed version in a `if:` step
* update the images used for the runner with a new version of `docker/docker compose`

Reply:
Have you tried specifying a different version such as `20.10.24` to the crazymax plugin instead of the defaulr (`latest`)?